### PR TITLE
Discover angi architecture

### PR DESCRIPTION
**Pull request by arbulu89**
# Description

Discovery of `angi` architecture HANA clusters.
A new field, `architecture_type` has been added to the hana cluster details, being "classic" and "angi" the possible values.

The `architecture_type` field is now available in the clusters api.

@abravosuse What do you think of the field name `architecture_type`?

## How was this tested?

UT added

### CODE REVIEW

File: lib/trento/discovery/policies/cluster_policy.ex [hunk 1]
```diff
@@ -269,7 +302,8 @@ defmodule Trento.Discovery.Policies.ClusterPolicy do
 
   defp parse_nodes_actual_roles(
```
Comment (CDimonaco) [inline lib/trento/discovery/policies/cluster_policy.ex:303]: maybe a comment to outline the 3 params

**Review by CDimonaco — APPROVED** [review]
Good job 🚀 

I left some comments, up to you, they are code style comments

File: lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex [hunk 2]
```diff
@@ -98,14 +116,37 @@ defmodule Trento.Discovery.Payloads.Cluster.ClusterDiscoveryPayload do
...
+      has_sap_hana_controller
+    )
   end
 
   defp parse_cluster_type(_), do: ClusterType.unknown()
 
-  defp do_detect_cluster_type(true, true, _), do: ClusterType.hana_scale_up()
-  defp do_detect_cluster_type(true, _, true), do: ClusterType.hana_scale_out()
-  defp do_detect_cluster_type(_, _, _), do: ClusterType.unknown()
+  defp parse_hana_glob_topology(%{
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex:129]: But this function doesn't return a boolean, it returns the value of `_glob_topology` if exists.
I will update the `has_hana_glob_topology` name in top, so it gets more logical

File: lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex [hunk 3]
```diff
@@ -66,6 +69,19 @@ defmodule Trento.Discovery.Payloads.Cluster.ClusterDiscoveryPayload do
     |> Map.put("additional_sids", parse_cluster_additional_sids(attrs))
   end
 
+  defp enrich_hana_architecture_type(%{"cluster_type" => cluster_type} = attrs)
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex:72]: I think I will leave this as is.
We already have some different ordering for the `enrich_cluster_sid` where we don't respect "perfect" ordering.
I prefer to at least keep the `enrich_` functions on top, and parsing afterwards.
Wdyt?

File: lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex [hunk 4]
```diff
@@ -98,14 +116,37 @@ defmodule Trento.Discovery.Payloads.Cluster.ClusterDiscoveryPayload do
         end)
       end)
 
-    do_detect_cluster_type(has_sap_hana_topology, has_sap_hana, has_sap_hana_controller)
+    do_detect_cluster_type(
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex:119]: If I understood correctly what you mean, you are saying to enrich the `architecture_type` first, and if it is angi, set the cluster type in the same action. And if not, do what we were doing before.
I initially tried that, but didn't get really good code.
But I see what you mean, and I will try back again, i might have better code now that I had when I initially started

File: lib/trento/discovery/policies/cluster_policy.ex [hunk 5]
```diff
@@ -167,7 +172,34 @@ defmodule Trento.Discovery.Policies.ClusterPolicy do
...
       nodes: parse_cluster_nodes(payload, sid),
       sbd_devices: parse_sbd_devices(sbd),
-      sites: parse_hana_scale_out_sites(cib, sid)
+      sites: parse_hana_scale_out_sites(HanaArchitectureType.classic(), cib, sid)
+    }
+  end
+
+  # angi architecture follows a really similar configuration to the scale out one.
+  # That's why some functions are shared with classic scale out setup
+  defp parse_cluster_details(
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/discovery/policies/cluster_policy.ex:181]: At the end, i was having already enough code duplication hehe
In any case, whoever looked for the proxy, would have the same questions, why angi is using scale out code XD

File: lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex]: code updated

File: lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex (hunk 4 above)
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex]: code updated

File: lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex]: code updated

File: lib/trento/discovery/policies/cluster_policy.ex (hunk 1 above)
Comment (arbulu89) [reply lib/trento/discovery/policies/cluster_policy.ex:306]: comment added

**Comment by arbulu89** [comment]
Hey @CDimonaco 
I have updated the payload code.
Now the `hana_architecture_type` and `cluster_type` are set together.
The code looks cleaner and we don't use that many functions.

**Comment by arbulu89** [comment]
Damn, clicked wrong button XD

File: lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex [hunk 6]
```diff
@@ -112,6 +133,24 @@ defmodule Trento.Discovery.Payloads.Cluster.ClusterDiscoveryPayload do
 
   defp do_detect_cluster_type(_), do: ClusterType.unknown()
 
+  defp put_hana_architecture(%{"cluster_type" => type} = attrs)
```
Comment (CDimonaco) [inline lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex:136]: I think we could rename this to put_hana_classic architecture, this clause is perfectly clear, but the other just put nothing, even a maybe_put_hana_classic_architecture but it's really a nitpick

**Review by CDimonaco — APPROVED** [review]
Beautiful! 

Thanks for addressing the feedbacks, I left a very little nitpick, as you prefer

LGTM! 🚀

File: lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex (hunk 6 above)
Comment (arbulu89) [reply lib/trento/discovery/payloads/cluster/cluster_discovery_payload.ex:136]: Deal!
I will rename to `maybe_put_hana_classic_architecture` and merge the PR

**Review by EMaksy — APPROVED** [review]
Hey man, code wise lgtm!

**Pull request merged**